apps/andi/lib/andi/harvest/harvester.ex

```diff
@@ -48,13 +66,43 @@ defmodule Andi.Harvest.Harvester do
 
   def dataset_update(datasets) do
     Enum.each(datasets, fn dataset ->
-      Brook.Event.send(instance_name(), dataset_update(), :andi, dataset)
+      case Organizations.get_harvested_dataset(dataset.id) do
+        %{include: false} -> Logger.info("Skipping dataset update for harvested dataset #{dataset.id}")
+        _ -> Brook.Event.send(instance_name(), dataset_update(), :andi, dataset)
+      end
     end)
   end
 
   def harvested_dataset_update(harvested_datasets) do
     Enum.each(harvested_datasets, fn harvested_dataset ->
-      Brook.Event.send(instance_name(), dataset_harvest_end(), :andi, harvested_dataset)
+      case Organizations.get_harvested_dataset(harvested_dataset["datasetId"]) do
+        %{include: false} -> Logger.info("Skipping dataset update for harvested dataset #{harvested_dataset["datasetId"]}")
+        _ -> Brook.Event.send(instance_name(), dataset_harvest_end(), :andi, harvested_dataset)
+      end
+    end)
+  end
+
+  defp remove_old_harvested_datasets(new_datasets, org_id) do
+    current_datasets = Organizations.get_all_harvested_datasets(org_id)
+
+    Enum.each(current_datasets, fn existing_dataset ->
```

Comment:
Would it be easier to have a function that overwrites the current harvested dataset instead of deleting the old one? That may not work for the ViewState as easily, but it would clean this up a little. If not, maybe break this block up into some helper functions?